spec/inputs/jms_spec.rb

```diff
@@ -0,0 +1,90 @@
+require "logstash/devutils/rspec/spec_helper"
+require "jms"
+
+def populate(queue_name, content)
+  require "logstash/event"
+
+  jms_config = {
+    :jndi_name => "/ConnectionFactory",
+    :jndi_context => {
+      'java.naming.factory.initial'=> 'org.jnp.interfaces.NamingContextFactory',
+      'java.naming.provider.url'=> 'jnp://localhost:1099',
+      'java.naming.factory.url.pkgs'=> 'org.jboss.naming:org.jnp.interfaces',
+      'java.naming.security.principal'=> 'guest',
+      'java.naming.security.credentials'=> 'guest'},
+    :require_jars => [
+      "/Applications/hornetq-2.4.0.Final/lib/hornetq-commons.jar",
+      "/Applications/hornetq-2.4.0.Final/lib/hornetq-core-client.jar",
+      "/Applications/hornetq-2.4.0.Final/lib/hornetq-jms-client.jar",
+      "/Applications/hornetq-2.4.0.Final/lib/jboss-jms-api.jar",
+      "/Applications/hornetq-2.4.0.Final/lib/jnp-client.jar",
+      "/Applications/hornetq-2.4.0.Final/lib/netty.jar"]
+  }
+
+  JMS::Connection.session(jms_config) do |session|
+    session.producer(:queue_name => queue_name) do |producer|
+      producer.send(session.message(content))
+    end
+  end
+end
+
+def process(pipeline, queue, content)
+  sequence = 0
+  Thread.new { pipeline.run }
+  event = queue.pop
+  pipeline.shutdown
+end # process
+
+describe "inputs/jms", :jms => true do
+
+
+  describe "read events from a queue" do
+    queue_name = "ExampleQueue"
```

Comment:
For the variable queue_name, content and config, we might use the rspec let helper methods as you can see in this link: https://www.relishapp.com/rspec/rspec-core/v/2-14/docs/helper-methods/let-and-let